### DISCUSSION TITLE
Modal Analysis and Component Mode Synthesis?

### DISCUSSION THREAD

**Discussion by oldninja** [idaholab/moose#20174] [Q&A Modules: Solid mechanics] [answered]
Dear MOOSE Team,

I am interested in performing modal analysis and component mode synthesis (CMS) using the moose framework. However, I looked for these terms in the documentation, and I could not find any information about them. That's why I opened this discussion to see if anyone could point me in the right direction. I'm more interested in performing the CMS part, sometimes referred as "superelement" in [NASTRAN](https://www.plm.automation.siemens.com/en_pl/Images/2652_tcm801-4486.pdf) or "substructure" in [Abaqus](https://abaqus-docs.mit.edu/2017/English/SIMACAEANLRefMap/simaanl-c-superelementdef.htm)/[CalculiX](https://www.feacluster.com/CalculiX/ccx_2.18/doc/ccx/node185.html) (more info [here](https://www.osti.gov/servlets/purl/1408346), [here](https://web.mit.edu/kjb/www/Principal_Publications/Component_mode_synthesis_with_subspace_iterations_for_controlled_accuracy_of_frequency_and_mode_shape_solutions.pdf), or [here](https://www.osti.gov/servlets/purl/1576135)). Any help would be greatly appreciated!

Thanks,

**Comment by GiudGiud** [comment, marked as answer]
@dschwen @somu15 do you know who might know about this?

**Reply by somu15** [reply]
Tagging @cbolisetti and @lynnmunday. This may have something to do with Craig Bampton as per one of the shared documents.

**Reply by oldninja** [reply]
> Craig Bampton

Yes, that is usually the underlying method either in Abaqus or Nastran.

**Reply by cbolisetti** [reply]
Yes, we have looked into implementing this in moose but never got funded. We don't have CMS/Craig Brampton right now, unfortunately.

**Reply by oldninja** [reply]
I see, thank you for letting me know.

**Comment by lynnmunday** [comment]
Here are some examples for modal anaylysis including eign analysis:
https://mooseframework.inl.gov/modules/solid_mechanics/1d_elastic_waves.html

If you're still interested in this, we have added some capabilities to do reduced order models that could probably be used to for super elements.
